File: go/ex2_1_2.go
package main

type EmailDst interface {
    sendEmail(content string)
}

type Person struct {}
func (p Person) sendEmail(content string) {}

func sendNewYearEmail(dst EmailDst) {
    dst.sendEmail("")
}

func main() {
    var pr Person = Person{}
    sendNewYearEmail(pr)
}

File: java/ex2_1_2.java
abstract class EmailDst {
    abstract void sendEmail(String content);
}

class Person extends EmailDst {
    void sendEmail(String content) {}
}

class Example {
    static void sendNewYearEmail(EmailDst dst) {
        dst.sendEmail("");
    }

    public static void main(String[] args) {
        Person pr = new Person();
        sendNewYearEmail(pr);
    }
}
